runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -819,27 +849,32 @@ ClientSessionData::getClassRecord(ClientSessionData::ClassInfo &classInfo)
 }
 
 const AOTCacheClassRecord *
-ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord)
+ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord, bool &uncachedClass)
    {
    TR_ASSERT(getROMMapMonitor()->owned_by_self(), "Must hold ROMMapMonitor");
 
    auto it = getROMClassMap().find(clazz);
    if (it == getROMClassMap().end())
+      {
+      uncachedClass = true;
       return NULL;
+      }
 
-   auto record = getClassRecord(it->second);
+   bool missingLoaderInfo = false;
+   auto record = getClassRecord(it->second, missingLoaderInfo);
    missingLoaderRecord = !record;
    return record;
    }
 
 const AOTCacheClassRecord *
-ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *stream)
+ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *stream, bool &missingLoaderInfo)
    {
    const AOTCacheClassRecord *record = NULL;
    bool missingLoaderRecord = false;
```

Comment:
Should simply pass down `missingLoaderInfo`. It will stay set to true if we request the missing info from the client and still fail.